.github/actions/lerna-version/action.yml

```diff
@@ -1,27 +1,38 @@
 name: Version packages
-description: Updates package version numbers for changed packages.
+description: Updates package version numbers and changelogs for changed packages.
 inputs:
-  update_type:
-    description: "Semver version update type"
+  release_type:
+    description: "Release type. If not 'stable', the current version will be bumped as a pre-release instead of a stable release."
     required: true
-    default: "prepatch"
-  prerelease_identifier:
-    description: "Prerelease identifier"
-    default: "experimental"
-    required: false
 outputs:
   VERSION:
     description: "The new latest version number across all packages"
     value: ${{ steps.lerna_version.outputs.VERSION }}
 runs:
   using: composite
   steps:
-    - name: Version packages
+    - name: Version packages & generate changelogs
       shell: bash
       id: lerna_version
-      run: ${{ github.action_path }}/version.sh "${{ inputs.update_type }}" "${{ inputs.prerelease_identifier }}"
+      run: ${{ github.action_path }}/version.sh ${{ inputs.release_type }}
 
-      # Lerna may introuce slight formatting changes to the package.json files
-    - name: Format files
+      # Lerna may introduce slight formatting changes to the package.json files.
+      # This can be removed once https://github.com/lerna/lerna/issues/4117 is closed.
+    - name: Apply formatting fixes
       shell: bash
       run: bun run check:fix
+
+    - name: Setup Git CLI
+      uses: ./.github/actions/setup-git
+      with:
+        name: ${{ github.actor }}
+
+    - name: Commit and tag v${{ steps.lerna_version.outputs.VERSION }}
+      shell: bash
+      run: |
+        git add .
+        git commit -m "chore: version bump to ${{ steps.lerna_version.outputs.VERSION }}"
+        # Create an annotated tag (`-m`) so that lerna can use it to determine which packages have changed
+        # See https://github.com/lerna/lerna/issues/1357#issuecomment-438162152
+        git tag "v${{ steps.lerna_version.outputs.VERSION }}" -m "v${{ steps.lerna_version.outputs.VERSION }}"
+        git push && git push --tags
```

Comment:
This was originally being called in the `tags` workflow that calls this action. It was moved into this action as I think there is no circumstance where one would want to version tags & create changelogs without also committing/pushing them, so it's better to abstract that away in the `tag` workflow and allow the versioning action to take care of its own side effects.